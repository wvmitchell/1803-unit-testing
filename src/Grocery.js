import React from 'react';

const Grocery = ({name, quantity, notes, starred, onStar, onPurchase, onDelete}) => (
  <div>
    <h3>{name}</h3>
    <p>Quantity: { quantity }</p>
    <p>{notes}</p>
    <button className="delete-button" onClick={onDelete}>Delete</button>
  </div>
)

export default Grocery
