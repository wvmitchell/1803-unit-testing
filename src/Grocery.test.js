import React from 'react'
import Grocery from './Grocery'
import { shallow } from 'enzyme'

describe('Grocery', () => {
  it.skip('should have a title in h3 tags', () => {
    // Setup
    const wrapper = shallow(<Grocery name="apples" starred={false} />)
    const expected = <h3>apples</h3>

    // Execution
    
    // Expectation or Assertion
    expect(wrapper.contains(expected)).toEqual(true)
  })

  it('should match the snapshot when all the info is passed', () => {
    const wrapper = shallow(<Grocery 
                              name="apples" 
                              quantity='10'
                              notes='apples are good'
                              starred={false} 
                              onStar={jest.fn()}
                              onDelete={jest.fn()}
                              onPurchase={jest.fn()}
                            />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should call onDelete when the delete button is clicked', () => {
    // Setup
    const mockDelete = jest.fn()
    const wrapper = shallow(<Grocery 
                              name="apples" 
                              quantity='10'
                              notes='apples are good'
                              starred={false} 
                              onStar={jest.fn()}
                              onDelete={mockDelete}
                              onPurchase={jest.fn()}
                            />)

    // Execution
    wrapper.find('.delete-button').simulate('click')

    // Expectation
    expect(mockDelete).toHaveBeenCalled()
  })
})
