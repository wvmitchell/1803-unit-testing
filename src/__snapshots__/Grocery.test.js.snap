// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grocery should match the snapshot when all the info is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grocery
    name="apples"
    notes="apples are good"
    onDelete={[Function]}
    onPurchase={[Function]}
    onStar={[Function]}
    quantity="10"
    starred={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          apples
</h3>,
        <p>
          Quantity: 
          10
</p>,
        <p>
          apples are good
</p>,
        <button
          className="delete-button"
          onClick={[Function]}
>
          Delete
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "apples",
        },
        "ref": null,
        "rendered": "apples",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Quantity: ",
            "10",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Quantity: ",
          "10",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "apples are good",
        },
        "ref": null,
        "rendered": "apples are good",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "className": "delete-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            apples
</h3>,
          <p>
            Quantity: 
            10
</p>,
          <p>
            apples are good
</p>,
          <button
            className="delete-button"
            onClick={[Function]}
>
            Delete
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "apples",
          },
          "ref": null,
          "rendered": "apples",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Quantity: ",
              "10",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Quantity: ",
            "10",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "apples are good",
          },
          "ref": null,
          "rendered": "apples are good",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "delete-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
